3drepo/3drepo.io
# ISSUE #2787 - Create Federation Modal

This fixes #2787 

#### Description
- Adds a modal for creating a federation
- This modal consists of two phases. One similar to the Edit Federation modal and one like the Federation Settings
- Adds API/Saga/Redux for creating Federation
- Currently the backend does not allow you to add containers to a federation


#### Test cases
- Create a federation with some containers
- Create a federation with no containers
- Go back and forth between modal phases and ensure data persists
- Close modal without submitting and the form will be reset
